x/observer/types/ballot_test.go

```diff
@@ -102,3 +102,56 @@ func TestVoter_IsBallotFinalized(t *testing.T) {
 		})
 	}
 }
+
+func Test_BuildRewardsDistribution(t *testing.T) {
+	tt := []struct {
+		name         string
+		voterList    []string
+		votes        []VoteType
+		ballotStatus BallotStatus
+		expectedMap  map[string]int64
+	}{
+		{
+			name:         "BallotFinalized_SuccessObservation",
+			voterList:    []string{"Observer1", "Observer2", "Observer3", "Observer4"},
+			votes:        []VoteType{VoteType_SuccessObservation, VoteType_SuccessObservation, VoteType_SuccessObservation, VoteType_FailureObservation},
+			ballotStatus: BallotStatus_BallotFinalized_SuccessObservation,
+			expectedMap: map[string]int64{
+				"Observer1": 1,
+				"Observer2": 1,
+				"Observer3": 1,
+				"Observer4": -1,
+			},
+		},
+		{
+			name:         "BallotFinalized_FailureObservation",
+			voterList:    []string{"Observer1", "Observer2", "Observer3", "Observer4"},
+			votes:        []VoteType{VoteType_SuccessObservation, VoteType_SuccessObservation, VoteType_FailureObservation, VoteType_FailureObservation},
+			ballotStatus: BallotStatus_BallotFinalized_FailureObservation,
+			expectedMap: map[string]int64{
```

Comment:
Maybe I'm mistaken but the votes don't seem to match the expected map. Was that intentional?

Reply:
Not sure I understand, the observer vote needs to match the finalized ballot status to get a rewards 
https://github.com/zeta-chain/zeta-node/blob/6f7159e349e856e6123f9b64b4491320b2b0821b/x/observer/types/ballot.go#L76-L76